src/compiler/checker.ts

```diff
@@ -6814,6 +6904,22 @@ module ts {
             return undefinedType;
         }
 
+        function checkAwaitExpression(node: AwaitExpression): Type {
+            // Grammar checking
+            if (!(node.parserContextFlags & ParserContextFlags.Await)) {
+                var parameter = getContainingParameter(node);
+                if (parameter && parameter.parserContextFlags & ParserContextFlags.Await) {
+                    grammarErrorAfterFirstToken(node, Diagnostics._0_expression_is_not_allowed_in_an_initializer, "await");

```

Comment:
I see, so just like 'yield', we allow the word 'await' as an identifier in these initializers. This code is just for the case when you had 'await expr'. That makes sense.

I don't think it makes a difference which of the two flags you check for.